src/Illuminate/Cache/RedisLock.php

```diff
@@ -48,7 +50,42 @@ public function acquire()
      */
     public function release()
     {
-        return (bool) $this->redis->eval(LuaScripts::releaseLock(), 1, $this->name, $this->owner);
+        $owner = $this->owner;
+
+        /* If a serialization mode such as "php" or "igbinary" and/or a
+         * compression mode such as "lzf" or "zstd" is enabled, the owner
+         * must be serialized and/or compressed by us, because phpredis does
+         * not do this for the eval command.
+         *
+         * Name must not be modified!
+         */
+        if ($this->isPhpRedis()) {
+            $owner = $this->redis->client()->_serialize($owner);
+
+            // Check first whether enabled at all before checking each as most likely it is not used.
+            if ($this->isPhpRedisCompressionEnabled()) {
+                if ($this->isPhpRedisLzfEnabled()) {
+                    $owner = \lzf_compress($owner);
+                } elseif ($this->isPhpRedisZstdEnabled()) {
+                    $owner = \zstd_compress(
+                        $owner,
+                        $this->redis->client()->getOption(constant('Redis::OPT_COMPRESSION_LEVEL'))
+                    );
+                } elseif ($this->isPhpRedisLz4Enabled()) {
+                    $owner = \lz4_compress(
+                        $owner,
+                        $this->redis->client()->getOption(constant('Redis::OPT_COMPRESSION_LEVEL'))
+                    );
+                } else {
+                    throw new UnexpectedValueException(sprintf(
+                        'Unknown phpredis compression in use (%d). Unable to release lock.',
+                        $this->redis->client()->getOption(constant('Redis::OPT_COMPRESSION'))
+                    ));
+                }
+            }
+        }
+
+        return (bool) $this->redis->eval(LuaScripts::releaseLock(), 1, $this->name, $owner);
```

Comment:
This `release()` method can be kept simple by moving the "phpredis" extension-specific code into another method:

```php
public function release()
{
    return (bool) $this->redis->eval(LuaScripts::releaseLock(), 1, $this->name, $this->phpRedisOwner() ?? $owner);
}

protected function phpRedisOwner()
{
    if (! $this->isPhpRedisCompressionEnabled()) {
        return;
    }

    $owner = $this->redis->client()->_serialize($this->owner);

    // etc.
}
```

Here's a Gist with an updated `RedisLock.php` class: https://gist.github.com/derekmd/039d1b7b83326fc69790552fb30c9cc8 Then a 220 line class is shortened to 181 lines.

Another option is to move these additions into a completely new class with a constructor that accepts dependencies `$this->owner` and `$this->redis->client()`.

Reply:
I have moved this whole code into a different class.

Reply:
Also I added a new test class for this new one. Seems to be a lot cleaner now as this code is very specific to the extension and should not be mixed with the general redis code.